### ISSUE TITLE
Type error with inheriting generic dataclass

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#1239]
## Environment data

-   Language Server version: Pylance language server 2020.12.1-pre.1 (pyright ea581b3f)
-   OS and version: Mac OS X 11.01
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.9 (brew)

## Expected behaviour

When inheriting a generic dataclass, the dataclass inheriting should inherit all the fields from the base class.

## Actual behaviour

Getting type error: Argument of type "Key" cannot be assigned to parameter "key" of type "Key" in function "__init__"
  Type "Key" cannot be assigned to type "Key"

<img width="954" alt="Screenshot 2020-12-03 at 19 19 35" src="https://user-images.githubusercontent.com/849479/101071280-a7201500-359c-11eb-8026-6731385a4aa8.png">

## Logs

```
[Info  - 7:43:16 AM] Pylance language server 2020.12.1-pre.1 (pyright ea581b3f) starting
[Info  - 7:43:16 AM] Server root directory: /Users/dbrattli/.vscode/extensions/ms-python.vscode-pylance-2020.12.1-pre.1/dist
[Error - 7:43:16 AM] stubPath /Users/dbrattli/Developer/Github/Expression/typings is not a valid directory.
[Info  - 9:27:27 AM] No configuration file found.
[Info  - 9:27:27 AM] Setting pythonPath for service "Expression": "/usr/local/opt/python@3.9/bin/python3"
```

## Code Snippet / Additional information

```python
from dataclasses import dataclass
from typing import Generic, TypeVar

Key = TypeVar("Key")
Value = TypeVar("Value")


@dataclass
class MapTreeLeaf(Generic[Key, Value]):
    key: Key
    value: Value


@dataclass
class MapTreeNode(MapTreeLeaf[Key, Value]):
    # key: Key
    # value: Value
    pass


def add(key: Key, value: Value):
    node: MapTreeNode[Key, Value] = MapTreeNode(key=key, value=value)
    ...
```

Type errors go away if you uncomment the key and value fields in the `MapTreeNode` class, but I think it should work without having to re-declare them. https://docs.python.org/3/library/dataclasses.html#inheritance

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pyright/pylance.

**Comment by erictraut** [comment]
This is now fixed in Pyright 1.1.92, which I just published. It will also be addressed in next week's Pylance release.
